Update config docs (#274)
trufflesuite/trufflesuite.com: src/docs/truffle/reference/configuration.md

If you specify `host` and `port` instead of `provider`, Truffle will create its own default HTTP provider using that host and port, and no minimal network connection will be opened, so there is no need to do the function wrapping workaround. That said, you wouldn't be able to use a custom provider in this case.

### contracts_directory

The default directory for uncompiled contracts is `./contracts` relative to the project root. If you wish to keep your contracts in a different directory you may specify a `contracts_directory` property.

Example:

> To have Truffle search recursively downward for contracts in `./allMyStuff/someStuff/theContractFolder` at compile time:

Reviewer: Is it really searching recursively downward, and grabbing all contracts at each level? Or is it just looking at the directed indicated?

Reviewer 2: Agreed that "recursively" might be a misnomer.

Author: In my toy app tests it searches down into other folders as well.

> To have Truffle search recursively downward for contracts in `./allMyStuff/someStuff/theContractFolder` at compile time:

Reviewer 2: ```suggestion
To have Truffle find contracts in `./allMyStuff/someStuff/theContractFolder` (recursively) at compile time:
```

Reviewer 2: Or
```suggestion
To have Truffle find contracts in `./allMyStuff/someStuff/theContractFolder` (and all subfolders) at compile time:
```
